### ISSUE TITLE
Switch to a FOSS license

### ISSUE THREAD

**Issue by suhr** [maidsafe/crust#131]
Subj.

**Comment by chandraprakash** [comment]
Thanks @suhr for the suggestion. We will consider all option available and review our License at later stage.

**Comment by ssokolow** [comment]
Be aware that discussion currently ongoing in /r/rust/ gives the impression that people who don't understand that "GPL trumps all" are being scared away by the current licensing situation.

To elaborate on that "GPL trumps all" bit...
1. The text of the GPL explicitly says that you may not apply restrictions beyond those imposed by the GPL itself (forbidding things like NDAs and royalties on GPL licensed source)
2. As a result, applying additional restrictions without modifying the text nullifies the GPL because users would be required to satisfy two directly conflicting rules simultaneously.
3. The text of the GPL is copyrighted to you and you are granted a license to to use it only if you leave it un-modified. (So, if you make a modified GPL to allow additional restrictions, you're guilty of copyright infringement and they can sue you.)
4. They explicitly require that the name "GPL" only be used for their license in the original form (I think that's enforced via trademark law)
5. Even if it were somehow possible to bind someone to "GPL + Restrictions" terms, the copyleft clause in the GPL would explicitly discard those additional restrictions, so I could ask my friend to get the code from you and then give it to me under the terms of a clean, unrestricted GPL license.

So, in short, the current situation:
1. Scares off developers who think it's possible to make a "GPL + Restrictions" license
2. Scares off people who know just enough about licensing to know that a modified GPL would be incompatible with ordinary GPL but not enough to know that adding new restrictions to the GPL is flat-out illegal.
3. Doesn't gain anything

If, as I suspect, you're saying that this is GPL **or** a custom license (MySQL-style), then you'll want to add a message to clarify that "GPL or, at your discretion..." part.

**Comment by dirvine** [comment]
Ah I see your point, we are working with some folks to ensure this is much clearer, already several law firms in London, Scotland and also USA, including meetings with GPL authors as well. We can surely make the distinction much clearer though as we progress. So GPL is very clearly pure GPL and can be forked used as pure GPL. I personally do prefer clean language so this makes sense. I also feel the PR contributor agreement we have will vanish very soon anyway as it causes more confusion than it helps at times. It was wiggle room for a license change to BSD/MIT etc. if we got into a corner, but it seems we will not get into such a corner anyway. 

Still a little way to go, but the main thing is the core libraries etc. should (I feel) be protected and truly Open, but perhaps after the API app devs make up their own mind (we do not enforce past API, obvious to many but not all, as you say). 

It is a constant item on our monthly meetings right now and will stay until we get the correct answer. As we are in mid launch now it should clear up very very soon as devs give us feedback (like this, so thank you)

Personally I have become a much bigger fan of GPL with all the recent issues and especially seeing some of the Open Source code from large companies that is just nonsense and not what they use internally. Our approach is everything we do is open and nothing is private. I now feel everyone in security code should do likewise, as I say that is personal though ;-). The company and community are much bigger and more important than me though.

**Comment by ssokolow** [comment]
I think the biggest problem right now is that...

1. You've got both a `COPYING` file and a `LICENSE` file and they're mutually incompatible.
2. People are likely to read one or the other but not both and then come to the wrong conclusion.
3. Less experienced people are more likely to read `LICENSE` first, which contains the non-GPL text.

If it's a case of "Available under your choice of the GPLv3 or our proprietary license", then clearly state that.

It'd probably also help to add a shields.io license badge to draw the eye to the terms.
![GPLv3](https://img.shields.io/crates/l/crust.svg)

**Comment by dirvine** [comment]
This is really neat and we will do this. We are sprint last day so @rossmuir will have this done next week. Than you very much for the help, very much appreciated.

**Comment by ssokolow** [comment]
Glad I could help. I collect lists of useful "free for open-source" services so, if there's anything else you might find useful (or if you'd just like me to continue listing things), feel free to ask.

Examples:

* [Trello-like UI for GitHub Issues](https://waffle.io/maidsafe/crust), 
* [commit message linter for pull requests](https://gitcop.com/)
* [Windows Travis-CI competitor](http://www.appveyor.com/),

**Comment by dirvine** [comment]
Thanks again and please shout whenever you see something uncomfortable and we will pay attention. So lets see how we fare next week to get at least a few steps further. Cheers -> 1 beer in kitty :-)

**Comment by dirvine** [comment]
We also use reviewable.io which is also free for open source and we found Jira and many others very happy to help.  Nice list though, I have signed up already for waffle.io as it looks great. We did use appveyor, but unfortunately it took to long to run our CI, so ended up switching to a local jenkins for this (and osx)

**Comment by ssokolow** [comment]
Reviewable.io is on my list. I just didn't want to dump an exhaustive list on you without being asked.

**Comment by vn971** [comment]
I would like to see the project as Free Software, too.

The issue I see though is that a license alone is _not_ enough to accomplish that.
There are lots of examples where a project just throws some small part away as GPL, selling proprietary versions at the same time, adding proprietary "plugins" and such. A seasoned user of GPL software will not go for this. A clearness on copyrights would definitely help in building the community and defining its goals.
